wiki/Community/Profile_badge/en.md

```diff
@@ -39,3 +39,4 @@ A side effect of this layout scheme was that the default, non-expanded height of
 
 - The first two profile badges were awarded to ::{ flag=PL }:: [niedzwiedz1124](https://osu.ppy.sh/users/9610) and ::{ flag=PL }:: [White Wolf](https://osu.ppy.sh/users/39828) on 6 September, 2009, for [winning *Tag Tournament*](https://osu.ppy.sh/community/forums/topics/17169).<!-- TODO: The changelog entry for badges being added is also on the same day, and should be mentioned. are these available online anywhere to link to? -clayton -->
 - Badges can link to other web pages for more context, like a tournament's forum page or a wiki entry.
+- Badges also protect all of the player usernames from being taken due to inactivity or them changing their current username.
```

Comment:
where does it say that their past usernames can't be taken?

Reply:
@Walavouchey it doesn't explicitly say that, but from personal experience I can say that for sure. My previous username could be taken before I got the badge, but now it says "already in use"

Reply:
`UsernameValidation::validateUsersOfUsername()` :stuck_out_tongue: 

the actual logic goes: if anyone who has the username or used to have the username also has a profile badge, then the username is locked

Reply:
actual code can be found here for future reference:

https://github.com/ppy/osu-web/blob/70d5ef18611069339f45e83cf26aaa3de8ee11f3/app/Libraries/UsernameValidation.php#L96-L99